shtdlib.sh

```diff
@@ -292,14 +295,21 @@ function readlink_m {
 }
 
 # Platform independent version sort
-# When input is piped it's assumed to be newline (NL) delimited
+# When input is piped it's assumed to be space and/or newline (NL) delimited
 # When passed as parameters each one is processed independently
 # shellcheck disable=2120
 function version_sort {
+    while read -rt 1 piped_data; do
+        declare -a piped_versions
```

Comment:
Don't you want this outside the while loop?

Reply:
Nope, you can't do it outside the loop because the first thing in the loop needs to be read (that way read gets the pipe), if I put the declaration outside then it gets the pipe.

I'll add a comment to the effect.

Reply:
Oh right.. so doesn't the array get re-declared? Is this statement a no-op if `piped_versions` is already defined?